src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -279,6 +281,186 @@ end
     return f
 end
 
+"""
+    flux_nonconservative_powell_local_symmetric(u_ll, u_rr,
+                                                orientation::Integer,
+                                                equations::IdealGlmMhdEquations2D)
+
+Non-symmetric two-point flux discretizing the nonconservative (source) term of
+Powell and the Galilean nonconservative term associated with the GLM multiplier
+of the [`IdealGlmMhdEquations2D`](@ref).
+
+This implementation uses a non-conservative term that can be written as the product
+of local and symmetric parts. It is equivalent to the non-conservative flux of Bohm
+et al. (`flux_nonconservative_powell`) for conforming meshes but it yields different
+results on non-conforming meshes(!).
+## References
+- Rueda-Ramírez, Gassner (2023). A Flux-Differencing Formula for Split-Form Summation By Parts
+  Discretizations of Non-Conservative Systems. https://arxiv.org/pdf/2211.14009.pdf.
+"""
+@inline function flux_nonconservative_powell_local_symmetric(u_ll, u_rr,
+                                                             orientation::Integer,
+                                                             equations::IdealGlmMhdEquations2D)
+    rho_ll, rho_v1_ll, rho_v2_ll, rho_v3_ll, rho_e_ll, B1_ll, B2_ll, B3_ll, psi_ll = u_ll
+    rho_rr, rho_v1_rr, rho_v2_rr, rho_v3_rr, rho_e_rr, B1_rr, B2_rr, B3_rr, psi_rr = u_rr
+
+    v1_ll = rho_v1_ll / rho_ll
+    v2_ll = rho_v2_ll / rho_ll
+    v3_ll = rho_v3_ll / rho_ll
+    v_dot_B_ll = v1_ll * B1_ll + v2_ll * B2_ll + v3_ll * B3_ll
+
+    # Powell nonconservative term:   (0, B_1, B_2, B_3, v⋅B, v_1, v_2, v_3, 0)
+    # Galilean nonconservative term: (0, 0, 0, 0, ψ v_{1,2}, 0, 0, 0, v_{1,2})
+    psi_avg = (psi_ll + psi_rr) #* 0.5 # We remove the 0.5 because the flux is always multiplied by 0.5
```

Comment:
If possible, could you mention the place/function where this multiplication takes place? The comment is already great, but adding a cross-reference might make it even better :-)

Reply:
This multiplication with 0.5 is done basically in all parts of Trixi where the non-conservative flux function is called. Examples:
https://github.com/trixi-framework/Trixi.jl/blob/c7dd7f5ee90168c08ee59bd22419627988e6a11b/src/solvers/dgsem_tree/dg_2d.jl#L325-L327

https://github.com/trixi-framework/Trixi.jl/blob/c7dd7f5ee90168c08ee59bd22419627988e6a11b/src/solvers/dgsem_tree/dg_2d.jl#L499-L504

https://github.com/trixi-framework/Trixi.jl/blob/c7dd7f5ee90168c08ee59bd22419627988e6a11b/src/solvers/dgsem_tree/dg_3d.jl#L373-L375

The flux that I added can also be called in all other parts of Trixi, so I made it consistent with the 0.5 multiplication. Should I explain that in more detail?

Reply:
No, but maybe you could add something like
```suggestion
    psi_avg = (psi_ll + psi_rr) #* 0.5 # The flux is already multiplied by 0.5 wherever the flux is used in the code
```
?